src/main/java/org/zalando/nakadi/validation/schema/CompatibilityModeChangeConstraint.java

```diff
@@ -14,7 +14,7 @@
     final Map<CompatibilityMode, List<CompatibilityMode>> allowedChanges = ImmutableMap.of(
             CompatibilityMode.COMPATIBLE, Lists.newArrayList(CompatibilityMode.COMPATIBLE),
             CompatibilityMode.FORWARD, Lists.newArrayList(CompatibilityMode.FORWARD, CompatibilityMode.COMPATIBLE),
-            CompatibilityMode.NONE, Lists.newArrayList(CompatibilityMode.NONE)
+            CompatibilityMode.NONE, Lists.newArrayList(CompatibilityMode.NONE, CompatibilityMode.FORWARD)
```

Comment:
@musiKk for that, you'll have to migrate from `none` to `forward` first, then from `forward` to `compatible`.